go-controller/pkg/ovn/master.go

```diff
@@ -533,29 +533,65 @@ func (oc *Controller) SetupMaster(masterNodeName string) error {
 	return nil
 }
 
-func addNodeLogicalSwitchPort(logicalSwitch, portName, portType, addresses, options string) (string, error) {
-	stdout, stderr, err := util.RunOVNNbctl("--", "--may-exist", "lsp-add", logicalSwitch, portName,
-		"--", "lsp-set-type", portName, portType,
-		"--", "lsp-set-options", portName, options,
-		"--", "lsp-set-addresses", portName, addresses)
+func addNodeLogicalSwitchPort(ovnNBClient goovn.Client, logicalSwitch, portName, portType, addresses string, options map[string]string) (string, error) {
+	var cmds []*goovn.OvnCommand
+	//var goOvnOptions map[string]string
+
+	lsp, err := ovnNBClient.LSPGet(portName)
+	if err != nil && err != goovn.ErrorNotFound {
```

Comment:
now I'm paranoid about this no schema thing :) We probably need to resolve that and have a common understanding of what errors we actually need to check for

Reply:
Makes sense..For now, I am falling back to the original pattern of checking for err.goovn.ErrorSchema. Hope that is OK.